### PR TITLE
Make car avoid objects from sides.

### LINKED ISSUE #23: App Design
As a driver I want to have a pretty interface and an easy to use, intuitive app. 

## Acceptance Criteria: 
- [x] Ensure that the colour scheme specified below are the only colours used in the app
- [x] Implement changes that have been voted on into the android app.
- [x] One user should test it and approve of the design.
- [x] Ensure that adjacent entities like buttons and background have distinct colours to ensure visibility
- [x] Add Intractable icons to make app use more Intuitive 

### HTML Hex Code for the project
|  **HEX CODE**  | **COLOR**  | 
| --- |--- |
|  000000 | Black  |   
|  14213D | Oxford Blue  |   
|  FCA311 | Orange Web  |
|  E5E5E5 | Platinum  |   
|  FFFFFF | White  |   

image

### LINKED ISSUE #48: Incoming avoidance from the sides
The simulated car will register when an object is sensed by either the side sensors on the car. When it senses an object that is on the side of the car it will speed up forward to try to avoid the object coming from the sides.

## Acceptance Criteria:

- [x] When the side of the car sensors registers an object it will perform an evasive maneuver. 
- [x] When evasive maneuver takes effect the car will speed forward to avoid the registered object on its side.
- [x] When the car no longer registers an object on its side it will continue to move forward one or two seconds.
- [x] When the car has finished it at evasive maneuver the car will come to a standstill waiting for the next command.

### PR DESCRIPTION
**Pull request by o55ian**
Make the car avoid objects coming from its side.

adding side avoidance method and calling it where need be.

close #48 

### :ballot_box_with_check: Definition of Done checklist
- [ ] Meaningful title and valuable description in pull request
- [ ] Code change tested manually if not covered by automated tests
- [ ] Pull request linked to one or more issues/stories
- [ ] Acceptance criteria of the linked stories satisfied
- [ ] Code change comprehended by those who approved it
- [ ] State of target branch maintained or improved after the code change

### CODE REVIEW

File: arduino/smartcar/vehicle_config.json [hunk 1]
```diff
@@ -1,4 +1,86 @@
 {
```
Comment (Valkama1) [inline arduino/smartcar/vehicle_config.json:1]: I think we removed this file in a previous PR, was adding it a mistake or is there a reason that it's back?

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -430,6 +431,16 @@ void incomingAvoidanceThreshold(long frontUltDis, long backUltDis)
     }
 }
 
+
+//allows vehicle to avoid objects on its side
+void directionalAvoidance(long leftUltDis,long rightUltDis){
```
Comment (Valkama1) [inline arduino/smartcar/smartcar.ino:436]: This logic will make the car drive indefinitely if it's standing next to a wall for example.

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -430,6 +431,16 @@ void incomingAvoidanceThreshold(long frontUltDis, long backUltDis)
...
+    if (leftUltDis <= 100 && leftUltDis != 0 || rightUltDis <= 100 && rightUltDis != 0){
+         car.setSpeed(100);
+            delay(300);
```
Comment (Valkama1) [inline arduino/smartcar/smartcar.ino:439]: Is there a way to avoid having this delay?

**Comment by Valkama1** [comment]
Why is this PR closing #23 ?

**Review by Valkama1 — CHANGES_REQUESTED** [review]
I don't think it's good that the car drives forever if it's next to a wall for example, you should try to check if this object is getting closer or not, and based on that move the car or not.

File: arduino/smartcar/vehicle_config.json (hunk 1 above)
Comment (o55ian) [reply arduino/smartcar/vehicle_config.json:1]: Dont know if its needed, Liam gave it to me when i was working on it and asked me to add it.

**Comment by Valkama1** [comment]
This PR does not actually fulfill the acceptance criteria of #48. So either update your code to do so, or, if the acceptance criteria aren't relevant, update the acceptance criteria of #48.

File: arduino/smartcar/vehicle_config.json (hunk 1 above)
Comment (Valkama1) [reply arduino/smartcar/vehicle_config.json:1]: Oh okay I see, my bad.

**Pull request closed without merging**